pilot/pkg/model/validation.go

```diff
@@ -1448,6 +1448,20 @@ func ValidateServiceRole(name, namespace string, msg proto.Message) error {
 		if len(rule.Services) == 0 {
 			errs = appendErrors(errs, fmt.Errorf("at least 1 service must be specified for rule %d", i))
 		}
+		// Regular rules and not rules (e.g. methods and not_methods should not be defined together).
+		sameAttributeKindError := "cannot have both regular and *not* attributes for the same kind (%s) for rule %d"
+		if len(rule.Hosts) > 0 && len(rule.NotHosts) > 0 {
+			errs = appendErrors(errs, fmt.Errorf(sameAttributeKindError, "i.e. hosts and not_hosts", i))
+		}
+		if len(rule.Methods) > 0 && len(rule.NotMethods) > 0 {
+			errs = appendErrors(errs, fmt.Errorf(sameAttributeKindError, "i.e. methods and not_methods", i))
+		}
+		if len(rule.Paths) > 0 && len(rule.NotPaths) > 0 {
```

Comment:
Similarly, paths and notPaths can be used together. You can define rules like, allow access to "/abc/*" except "/abc/private".